src/core_plugins/timelion/public/directives/timelion_expression_input_helpers.js

```diff
@@ -52,13 +69,69 @@ export class FunctionSuggestions {
   }
 }
 
-export function suggest(expression, functionList, Parser) {
+function inLocation(cursorPosition, location) {
+  return cursorPosition >= location.min && cursorPosition <= location.max;
+}
+
+function extractSuggestionsFromParsed(result, cursorPosition, functionList) {
+  const activeFunc = result.functions.find((func) => {
+    return cursorPosition >= func.location.min && cursorPosition < func.location.max;
+  });
+
+  if (activeFunc) {
+    const funcDefinition = functionList.find((func) => {
+      return func.name === activeFunc.function;
+    });
+    const alreadyProvidedArguments = activeFunc.arguments.map((arg) => {
+      return arg.name;
+    });
+
+    // return function suggestion if cursor is outside of parentheses
+    // location range includes '.', function name, and '('.
+    const openParen = activeFunc.location.min + activeFunc.function.length + 2;
+    if (cursorPosition < openParen) {
+      return { list: [funcDefinition], location: activeFunc.location, type: SUGGESTION_TYPE.FUNCTIONS };
+    }
+
+    const args = funcDefinition.chainable ? funcDefinition.args.slice(1) : funcDefinition.args.slice(0);
```

Comment:
I'm dumb... could you add a brief comment to illuminate the reason behind slicing from 1 or 0?

Reply:
Test https://github.com/nreese/kibana/blob/6441723ccddb6e4b5253219607b32b23be254be1/src/core_plugins/timelion/public/directives/__tests__/timelion_expression_input_helpers.js#L121 should help - but the goal is to not show the inputSeries argument in the suggestion list when the function is chainable. I will add a comment